### PR TITLE
Fix cancel an in progress command ref #150

### LINKED ISSUE #150: [WIP]Fix cancel an in progress command.
**Describe the bug**
https://github.com/skycoin/hardware-wallet/issues/321

**Environment information:**

Any

**Steps to Reproduce**

https://github.com/skycoin/hardware-wallet/issues/321

**Actual behavior**
Unable to cancel backup process.

**Expected behavior**
Allow to cancel backup.

**Additional context**
https://github.com/skycoin/hardware-wallet/issues/321

### PR DESCRIPTION
**Pull request by stdevAlDen**
Fixes #150

Changes:
- Make `NewDevice` a singleton
- Enforce `Connect` requests to be sequential.
- Use a timeout for data transfer in `libbusb` (GNU/Linux).

Does this change need to mentioned in CHANGELOG.md?
yes

Requires changes in protobuff specs?
no

Requires testing
no

Comments about testing , should you have some

### CODE REVIEW

File: src/skywallet/skywallet.go [hunk 1]
```diff
@@ -132,30 +144,29 @@ func (d *Device) Close() {
 // Connect makes a connection to the connected device
 func (d *Device) Connect() error {
 	d.Lock()
-	defer d.Unlock()
 	// close any existing connections
 	if d.dev != nil {
+		return errors.New("already connected")
 		d.dev.Close(false)
 		d.dev = nil
```
Comment (therealssj) [inline src/skywallet/skywallet.go]: the next two lines are never reached

**Comment by stdevAlDen** [comment]
@therealssj can you please make a review?
I want to make sure this doesn't break something on your side

There are some changes remaining, but I thing that these are the mos critical (major changes) for now.

File: src/skywallet/skywallet.go (hunk 1 above)
Comment (stdevAlDen) [reply src/skywallet/skywallet.go]: > the next two lines are never reached

@therealssj yeah you are right, I'm going to remove the possibility to cancel a current connection when you call cancel again

**Comment by therealssj** [comment]
@stdevAlDen update?

**Comment by olemis** [comment]
@therealssj there is a tiny issue @stdevAlden still debugging teh patch to fix it

File: src/skywallet/usb/libusb.go [hunk 2]
```diff
@@ -432,8 +432,7 @@ func (d *LibUSBDevice) readWrite(buf []byte, endpoint uint8) (int, error) {
 		}
 
 		d.transferMutexLock()
-		// This has no timeout, but is stopped by Cancel_Sync_Transfers_On_Device
-		p, err := lowlevel.Interrupt_Transfer(d.dev, endpoint, buf, 0)
+		p, err := lowlevel.Interrupt_Transfer(d.dev, endpoint, buf, 250)
```
Comment (olemis) [inline src/skywallet/usb/libusb.go]: Use a constant

**Pull request merged**